src/main/rx/cc2500_frsky_x.c

```diff
@@ -276,9 +277,16 @@ void frSkyXSetRcData(uint16_t *rcData, const uint8_t *packet)
         } else {
             j = 0;
         }
+        if (c[i] == 0)
+        {
+            continue;
```

Comment:
I was seeing a lot of single channel glitches going to the min value in the rcdata.  For some reason the packet passes the CRC with those 0 values, so I assume it is legit values getting passed up and the frame is valid... but the 0 values that aren't valid and are the causes of the twitches.   If you remove that and look at the receiver data in the configurator you can see the glitches.

Wish I had SWD to be able to hook up to one of these...

I had even run the CRC packet check just above this and was still getting good packets.  I was making sure there wasn't an overflow or a stray pointer.

I first tried it because I had noticed that in the arducopter implementation: https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Radio/AP_Radio_cc2500.cpp

Reply:
Ok, that makes me wondering if this isn't caused by a broken TX implementation (which could also explain why I don't see these glitches). What TX do you use?

Even if just ignoring the glitches is the right thing to do, I think instead of keeping certain channels at their old values, and updating others, the correct approach is to either use or discard the entire frame. Particularly for channels that are used for modes, partial updates might lead to non-desired combinations with negative effects.

Reply:
This is with a Taranis x9d plus, so I'd hope it would be right! Although I have been having really bad range (100s of feet) with my telemetry link on all my quads, although RC range is fine.  Could my my RF module is not happy.

I also have a multi-module I could give a try to see if that does it as well.

Reply:
It's weird. And I'm not saying that detecting the 0 channels isn't part of the fix, but to me it looks like this is just a symptom, and we should try and find the root cause for it.